examples/subscription/src/main_subscription.cpp

```diff
@@ -35,164 +35,263 @@ using namespace uprotocol::uuid;
 using namespace uprotocol::v1;
```

Comment:
You are missing the test that will check the actual notification logic.
For example - 
1. send subscription request for a remote topic
2. server returns SUBSCRIBE_PENDING
3. As we dont have remote subscriptions support yet, the server should wait for one second, and send notification SUBSCRIBED (this part should be simulated in the server implementation)